test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,434 @@
+        assertNull(reader0.readNextEvent(500).getEvent());
+        reader0.close();
+
+        //Scaling up begin
+        Map<Double, Double> keyRanges = new HashMap<>();
+        keyRanges.put(0.0, 0.5);
+        keyRanges.put(0.5, 1.0);
+
+        Boolean status = controller.scaleStream(stream,
+                Collections.singletonList(0L),
+                keyRanges,
+                executor).getFuture().get();
+        assertTrue(status);
+
+        write30ByteEvents(5, writer);
+
+        ArrayList<SegmentRange> rangeList = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        List<Segment> allSegmentList = rangeList.stream().map(SegmentRange::getSegment).collect(Collectors.toList());
+        assertEquals(3, allSegmentList.size());
+        log.info("After scale up all the segment list : {}", allSegmentList);
+
+        Segment segment1 = Segment.fromScopedName(streamScope + "/" + streamName + "/1.#epoch.1");
+        Segment segment2 = Segment.fromScopedName(streamScope + "/" + streamName + "/2.#epoch.1");
+        log.info("Segment1 name {} and Segment2 name {}", segment1.getScopedName(), segment2.getScopedName());
+
+        long approxDistanceToNextOffset = 180L;
+        StreamCut nextStreamCut2 = batchClient.getNextStreamCut(streamCut1, approxDistanceToNextOffset);
+        log.info("Next stream cut2 {}", nextStreamCut2);
+
+        assertTrue(nextStreamCut2 != null);
+        assertTrue(nextStreamCut2.asImpl().getPositions().size() == 2);
+        assertTrue(nextStreamCut2.asImpl().getPositions().containsKey(segment1) &&
+                nextStreamCut2.asImpl().getPositions().containsKey(segment2));
+        //Scaling up end
+
+        //Scaling down start
+        ArrayList<Long> toSeal = new ArrayList<>();
+        toSeal.add(computeSegmentId(1, 1));
+        toSeal.add(computeSegmentId(2, 1));
+
+        Map<Double, Double> keyRanges1 = new HashMap<>();
+        keyRanges1.put(0.0, 1.0);
+
+        Boolean status1 = controller.scaleStream(stream,
+                Collections.unmodifiableList(toSeal),
+                keyRanges1,
+                executor).getFuture().get();
+        assertTrue(status1);
+
+        write30ByteEvents(5, writer);
+
+        Segment segment3 = Segment.fromScopedName(streamScope + "/" + streamName + "/3.#epoch.2");
+        log.info("segment3 name :{}", segment3.getScopedName());
+
+        ArrayList<SegmentRange> segmentList2 = Lists.newArrayList(
+                batchClient.getSegments(Stream.of(streamScope, streamName), StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        log.info("Segment List2 :{}", segmentList2);
+
+        Map<Segment, Long> map1 = segmentList2.stream().collect(Collectors.toMap(SegmentRange::getSegment, value -> value.getEndOffset()));
+        StreamCut streamCut = new StreamCutImpl(Stream.of(streamScope, streamName),
```

Comment:
why you are constructing this streamcut? Once all data been read from currentsegments ie segment1 and segement2 when you scale down segment then call getNextStreamcut(**nextStreamCut2**, limit)